src/qt/transactionrecord.cpp

```diff
@@ -53,30 +53,56 @@ QList<TransactionRecord> TransactionRecord::decomposeTransaction(const CWallet *
     }
 
     if (wtx.IsZerocoinSpend() || isAllSigmaSpendFromMe) {
+        CAmount nTxFee = nDebit - wtx.GetValueOut();
+        bool first = true;
+
+        bool isAllToMe = true;
         for (const CTxOut& txout : wtx.vout) {
-            if (wtx.IsChange(txout)) {
-                continue;
+            if (!wallet->IsMine(txout)) {
+                isAllToMe = false;
+                break;
             }
-            isminetype mine = wallet->IsMine(txout);
+        }
 
+        if(isAllToMe){
             TransactionRecord sub(hash, nTime);
             CTxDestination address;
+            sub.involvesWatchAddress = ISMINE_WATCH_ONLY;
+            if (ExtractDestination(wtx.vout[0].scriptPubKey, address) && IsMine(*wallet, address)) {
+                sub.type = TransactionRecord::SpendToSelf;
+                sub.address = CBitcoinAddress(address).ToString();
+                sub.credit = -nTxFee;
+                parts.append(sub);
+            }
+        } else {
+            for (const CTxOut& txout : wtx.vout) {
+                if (wtx.IsChange(txout)) {
+                    continue;
+                }
+                isminetype mine = wallet->IsMine(txout);
 
-            if (mine) {
-                sub.involvesWatchAddress = mine & ISMINE_WATCH_ONLY;
-                if (ExtractDestination(txout.scriptPubKey, address) && IsMine(*wallet, address))
-                {
-                    sub.type = TransactionRecord::SpendToSelf;
+                TransactionRecord sub(hash, nTime);
+                CTxDestination address;
+
+                if (mine) {
+                    sub.involvesWatchAddress = mine & ISMINE_WATCH_ONLY;
+                    if (ExtractDestination(txout.scriptPubKey, address) && IsMine(*wallet, address)) {
+                        sub.type = TransactionRecord::SpendToSelf;
+                        sub.address = CBitcoinAddress(address).ToString();
+                        sub.credit = txout.nValue;
+                        parts.append(sub);
```

Comment:
We forgot to set `sub.idx`.

Reply:
We were not setting it previously too.

Reply:
Yes. It is a bug from the previous code. Could you fix it with this PR? This PR already working on this part so we don't need another PR to fix it.